WordPress/Classes/ViewRelated/Media/WPAndDeviceMediaLibraryDataSource.m

```diff
@@ -13,28 +13,71 @@ @implementation WPAndDeviceMediaLibraryDataSource
 
 - (instancetype)initWithBlog:(Blog *)blog
 {
+    return [self initWithBlog:blog dataSourceType:MediaPickerDataSourceTypeDevice];
+}
+
+- (instancetype)initWithBlog:(Blog *)blog
+              dataSourceType:(MediaPickerDataSourceType)sourceType
+{
     self = [super init];
     if (self) {
         _mediaLibraryDataSource = [[MediaLibraryPickerDataSource alloc] initWithBlog:blog];
```

Comment:
Why not to move the init of the `_mediaLibraryDataSource` to the `commonInitWithSourceType` method.